test/recipes/81-test_cmp_cli.t

```diff
@@ -0,0 +1,294 @@
+    my $str = shift;
+    $str =~ s/^\"(.*?)\"$/$1/;
+    return $str;
+}
+
+my $proxy = "<EMPTY>";
+$proxy = chop_dblquot($ENV{http_proxy} // $ENV{HTTP_PROXY} // $proxy);
+$proxy =~ s{^https?://}{}i;
+my $no_proxy = $ENV{no_proxy} // $ENV{NO_PROXY};
+
+my $app = "apps/openssl cmp";
+
+my @cmp_basic_tests = (
+    [ "show help",                        [ "-config", '""', "-help"               ], 0 ],
+    [ "CLI option not starting with '-'", [ "-config", '""',  "days", "1"          ], 1 ],
+    [ "unknown CLI option",               [ "-config", '""', "-dayss"              ], 1 ],
+    [ "bad int syntax: non-digit",        [ "-config", '""', "-days", "a/"         ], 1 ],
+    [ "bad int syntax: float",            [ "-config", '""', "-days", "3.14"       ], 1 ],
+    [ "bad int syntax: trailing garbage", [ "-config", '""', "-days", "314_+"      ], 1 ],
+    [ "bad int: out of range",            [ "-config", '""', "-days", "2147483648" ], 1 ],
+);
+
+my $rsp_cert = "signer_only.crt";
+my $outfile = "test.cert.pem";
+my $secret = "pass:test";
+my $localport = 1700;
+
+# this uses the mock server directly in the cmp app, without TCP
+sub use_mock_srv_internally
+{
+    ok(run(cmd([bldtop_dir($app),
+                "-config", '""',
+                "-use_mock_srv", "-srv_ref", "mock server",
+                "-srv_cert",  "server.crt", # used for setting sender
+                "-srv_secret", $secret,
+                "-poll_count", "1",
+                "-rsp_cert", $rsp_cert,
+                "-cmd", "cr",
+                "-subject", "/CN=any",
+                "-newkey", "signer.key",
+                "-recipient", "/O=openssl_cmp", # if given must be consistent with sender
+                "-secret", $secret,
+                "-ref", "client under test",
+                "-certout" , $outfile]))
+       && compare_text($outfile, $rsp_cert) == 0,
+       "CMP app with -use_mock_srv and -poll_count 1");
+    unlink $outfile;
+}
+
+# the CMP server configuration consists of:
+my $ca_dn;      # The CA's Distinguished Name
+my $server_dn;  # The server's Distinguished Name
+my $server_cn;  # The server's domain name
+my $server_ip;  # The server's IP address
+my $server_port;# The server's port
+my $server_tls; # The server's TLS port, if any, or 0
+my $server_path;# The server's CMP alias
+my $server_cert;# The server's cert
+my $kur_port;   # The server's port for kur (cert update)
+my $pbm_port;   # The server port to be used for PBM
+my $pbm_ref;    # The reference for PBM
+my $pbm_secret; # The secret for PBM
+my $column;     # The column number of the expected result
+my $sleep = 0;  # The time to sleep between two requests
+
+# The local $server_name variables below are among others taken as the name of a
+# sub-directory with server-specific certs etc. and CA-specific config section.
+
+sub load_config {
+    my $server_name = shift;
+    my $section = shift;
+    my $test_config = $ENV{CMP_CONFIG} // "test_$server_name.cnf";
```

Comment:
Is the environment variable here for debug purposes? I guess it could cause the tests to fail if it happens to be set in the environment. Should it have an OPENSSL prefix?

Reply:
The env. var. is meant for use with other config file types, but could be also used for debugging.
I've added `OPENSSL_` to all `CMP_*` environment variables used there.